### PR TITLE
feat(web-client): Hide XRP Balance

### PR DESCRIPTION
**Pull request by IscoRuta98**
## Aim: Hide XRP balance for various organizations.
Acceptance criteria: [Clickup](https://app.clickup.com/t/2y436yt)

### Tasks

- [x] Use feature variable to hide XRP balance on relevant pages.
- [x] Use feature variable to hide XRP Payment Option.
- [x] Use feature variable for the user to delete wallet by sending issued currency (not XRP).
- [x] Update PR with latest changes made to main branch.

### CODE REVIEW

File: web-client/src/environments/environment.staging.ts [hunk 1]
```diff
@@ -24,5 +24,7 @@ export const environment: Environment = {
   },
   commissionPercentage: 0.01,
   tokenIssuer: 'rpJv16Qmn2rQP6UC6UFsNRnVy5arkQihPP',
+  xrpIssuer: 'rpJv16Qmn2rQP6UC6UFsNRnVy5arkQihPP',
+  hideXrpBalance: false,
```
Comment (billguo99) [inline web-client/src/environments/environment.staging.ts:28]: keep order of variables the same across environment files

File: web-client/src/environments/environment.prod.ts [hunk 2]
```diff
@@ -24,5 +24,7 @@ export const environment: Environment = {
   },
   commissionPercentage: 0.01,
   tokenIssuer: 'rpJv16Qmn2rQP6UC6UFsNRnVy5arkQihPP',
+  xrpIssuer: 'rpJv16Qmn2rQP6UC6UFsNRnVy5arkQihPP',
+  hideXrpBalance: false,
```
Comment (billguo99) [inline web-client/src/environments/environment.prod.ts:28]: keep order of variables the same across environment files

File: web-client/src/environments/environment.palau-staging.ts [hunk 3]
```diff
@@ -25,4 +25,6 @@ export const environment: Environment = {
   commissionPercentage: 0.01,
   tokenIssuer: 'rngdKd8BAM3etQcb12DvGxd5Ps9MocAvPa',
   tokenSymbol: 'PSC',
+  hideXrpBalance: true,
+  xrpIssuer: 'rngdKd8BAM3etQcb12DvGxd5Ps9MocAvPa',
```
Comment (billguo99) [inline web-client/src/environments/environment.palau-staging.ts:29]: xrpIssuer move above hidexrpbalance

File: web-client/src/environments/environment.bhutan-staging.ts [hunk 4]
```diff
@@ -25,4 +25,6 @@ export const environment: Environment = {
   commissionPercentage: 0.01,
   tokenIssuer: 'rKitZq2qY8REiq7xvY3MfaqMhQkMekJtK',
   tokenSymbol: 'BTN',
+  hideXrpBalance: true,
+  xrpIssuer: 'rKitZq2qY8REiq7xvY3MfaqMhQkMekJtK',
```
Comment (billguo99) [inline web-client/src/environments/environment.bhutan-staging.ts:29]: move up

File: web-client/src/app/views/delete-user/delete-user.page.ts [hunk 5]
```diff
@@ -40,18 +44,47 @@ export class DeleteUserPage implements OnInit {
...
+  hideXrpBalance = environment.hideXrpBalance;
+
+  tokenIssuer = environment.tokenIssuer;
+
+  tokenSign: string | undefined;
+
+  /** A balance held by the current user. */
+  balances = this.sessionQuery.allBalances;
+
+  paymentOptions?: PaymentOption[];
```
Comment (billguo99) [inline web-client/src/app/views/delete-user/delete-user.page.ts:56]: is this used anywhere?

File: web-client/src/app/views/delete-user/delete-user.page.ts [hunk 6]
```diff
@@ -40,18 +44,47 @@ export class DeleteUserPage implements OnInit {
   /** @see validatedAddress */
   address?: string;
 
+  hideXrpBalance = environment.hideXrpBalance;
```
Comment (billguo99) [inline web-client/src/app/views/delete-user/delete-user.page.ts:47]: can just use environment.hideXrpBalance as is, don't need to create separate variable

File: web-client/src/app/views/delete-user/delete-user.page.ts [hunk 7]
```diff
@@ -40,18 +44,47 @@ export class DeleteUserPage implements OnInit {
...
+
+  tokenIssuer = environment.tokenIssuer;
```
Comment (billguo99) [inline web-client/src/app/views/delete-user/delete-user.page.ts:49]: same here, don't need another variable

File: web-client/src/app/views/delete-user/delete-user.page.ts [hunk 8]
```diff
@@ -40,18 +44,47 @@ export class DeleteUserPage implements OnInit {
...
+    } else {
+      this.tokenSign = 'XRP';
+    }
+
+    this.sessionQuery.allBalances.subscribe((balances) => {
+      this.checkAmountIsZero = balances.some(
+        (amount) =>
+          amount.assetDisplay.assetSymbol === 'FOO' && amount.amount !== 0
+      );
+      console.log(this.checkAmountIsZero);
```
Comment (billguo99) [inline web-client/src/app/views/delete-user/delete-user.page.ts:86]: remove this console line

File: web-client/src/app/views/delete-user/delete-user.page.ts [hunk 9]
```diff
@@ -40,18 +44,47 @@ export class DeleteUserPage implements OnInit {
...
+    if (this.hideXrpBalance) {
+      this.tokenSign = environment.tokenSymbol;
+    } else {
+      this.tokenSign = 'XRP';
+    }
+
+    this.sessionQuery.allBalances.subscribe((balances) => {
+      this.checkAmountIsZero = balances.some(
+        (amount) =>
+          amount.assetDisplay.assetSymbol === 'FOO' && amount.amount !== 0
```
Comment (billguo99) [inline web-client/src/app/views/delete-user/delete-user.page.ts:84]: hardcoded `FOO`, do you have to change this?

File: web-client/src/app/views/delete-user/delete-user.page.html [hunk 10]
```diff
@@ -7,24 +7,47 @@
   </ion-toolbar>
 </ion-header>
 
-<ion-content>
+<ion-content *ngIf="hideXrpBalance; else Default">
   <div class="ion-text-center p-2">
     <p>
-      You will need to enter a wallet address to send your remaining XRP balance
-      to.
+      You will need to send your remaining {{tokenSign}} balance before you can
```
Comment (billguo99) [inline web-client/src/app/views/delete-user/delete-user.page.html:13]: will make sense to show this message when the digital asset amount is not zero

File: web-client/src/app/views/delete-user/delete-user.page.ts [hunk 11]
```diff
@@ -40,18 +44,47 @@ export class DeleteUserPage implements OnInit {
...
     private navCtrl: NavController,
     private modalCtrl: ModalController,
     public sessionService: SessionService,
     public sessionQuery: SessionQuery,
     public notification: SwalHelper,
     public sessionAlgorandService: SessionAlgorandService,
-    public sessionXrplService: SessionXrplService
+    public sessionXrplService: SessionXrplService,
+    public deleteUserSrvice: DeleteUserService,
+    public purePayPageComponent: PurePayPageComponent
```
Comment (billguo99) [inline web-client/src/app/views/delete-user/delete-user.page.ts:69]: purePayPageComponent not used anywhere so can remove all of it from your delete files

**Review by billguo99 — CHANGES_REQUESTED** [review]

File: web-client/src/app/components/pure-pay-page/pure-pay-page.component.ts [hunk 12]
```diff
@@ -65,9 +66,17 @@ export class PurePayPageComponent implements OnInit, OnChanges {
...
+      filteredPaymentOptions = this.paymentOptions?.filter(
+        (paymentOption) =>
+          paymentOption.senderBalance.assetDisplay.assetSymbol !== 'XRP'
+      );
+      this.paymentOptions = filteredPaymentOptions;
+    }
   }
 
-  private getPaymentOptions(): PaymentOption[] | undefined {
+  public getPaymentOptions(): PaymentOption[] | undefined {
```
[the remark this answers is not in the document]
Comment (IscoRuta98) [reply web-client/src/app/components/pure-pay-page/pure-pay-page.component.ts:79]: Ah yes, thanks for catching this. I forgot to do this after I was able to hide xrp payments.

File: web-client/src/app/views/delete-user/delete-user.page.ts (hunk 9 above)
Comment (IscoRuta98) [reply web-client/src/app/views/delete-user/delete-user.page.ts:84]: Thanks for catching this; it should be `environment.tokenSymbol`.

File: web-client/src/app/views/delete-user/delete-user.page.ts (hunk 11 above)
Comment (IscoRuta98) [reply web-client/src/app/views/delete-user/delete-user.page.ts:69]: Thanks!

File: web-client/src/app/views/delete-user/delete-user.page.ts (hunk 6 above)
Comment (IscoRuta98) [reply web-client/src/app/views/delete-user/delete-user.page.ts:47]: I created a separate variable to use `hideXrpBalance` in [delete-user.page.html](https://github.com/ntls-io/nautilus-wallet/blob/aaef347fe516a445a01d9ed6e577beed5afd5441/web-client/src/app/views/delete-user/delete-user.page.html#L10). I tried using `environment.hideXrplBalance` inside the ngIf directive, but it looks like I need to initialize the variable inside delete-user.page.ts.

File: web-client/src/app/views/delete-user/delete-user.page.ts (hunk 5 above)
Comment (IscoRuta98) [reply web-client/src/app/views/delete-user/delete-user.page.ts:56]: `tokenSign `is used in [delete-user.page.html](https://github.com/ntls-io/nautilus-wallet/blob/aaef347fe516a445a01d9ed6e577beed5afd5441/web-client/src/app/views/delete-user/delete-user.page.html#L13). However `balances` and paymentOptions are not.

File: web-client/src/app/views/delete-user/delete-user.page.html (hunk 10 above)
Comment (IscoRuta98) [reply web-client/src/app/views/delete-user/delete-user.page.html:13]: Thanks! In the latest commit, I've added another message to show when the issued currency balance is zero.

**Comment by billguo99** [comment]
I see you've made the xrpIssuer the same as the tokenIssuer, this will probably change, just a note.

**Review by billguo99 — COMMENTED** [review]
Why did you create a Delete-User service? Did you consider just writing the deleteWallet() function inside delete-user.page.ts?

File: web-client/src/app/views/delete-user/delete-user.page.ts [hunk 13]
```diff
@@ -40,18 +42,39 @@ export class DeleteUserPage implements OnInit {
...
   constructor(
     private navCtrl: NavController,
     private modalCtrl: ModalController,
     public sessionService: SessionService,
     public sessionQuery: SessionQuery,
     public notification: SwalHelper,
     public sessionAlgorandService: SessionAlgorandService,
-    public sessionXrplService: SessionXrplService
+    public sessionXrplService: SessionXrplService,
+    public deleteUserSrvice: DeleteUserService
```
[the remark this answers is not in the document]
Comment (IscoRuta98) [reply web-client/src/app/views/delete-user/delete-user.page.ts]: Thanks!

**Comment by IscoRuta98** [comment]
> Why did you create a Delete-User service? Did you consider just writing the deleteWallet() function inside delete-user.page.ts?

@billguo99, I did, however I thought it'd be efficient to write the function as a service rather incase it is used elsewhere in the future.

**Comment by billguo99** [comment]
It's already used elsewhere on the deposit-funds.page. I think perhaps get Jonathan's opinion on this as he would probably know best on how to set this up in one place and how it should look like.

**Comment by IscoRuta98** [comment]
> It's already used elsewhere on the deposit-funds.page. I think perhaps get Jonathan's opinion on this as he would probably know best on how to set this up in one place and how it should look like.

@jongbonga, I would like your input with how I went about creating another deleteWallet() function by vreating a deleteService.service. The reason being is that the deleteWallet() function in the deposit-funds.page. uses a receiver's address inputted by the user, and the notification message for a successful deletion is different between the pilot environments and a 'normal' environment. Is what I have done fine, or would you suggest sticking to one deleteWallet() function.

**Pull request merged**